services/thermal_mgr/thermal_mgr.c

```diff
@@ -17,55 +18,107 @@ static StackType_t thermalMgrTaskStack[THERMAL_MGR_STACK_SIZE];
 
 #define THERMAL_MGR_QUEUE_LENGTH 10U
 #define THERMAL_MGR_QUEUE_ITEM_SIZE sizeof(thermal_mgr_event_t)
+#define THERMAL_MGR_HYSTERESIS_TEMP 75U
+#define THERMAL_MGR_OVERTEMP 80U
 
 static QueueHandle_t thermalMgrQueueHandle;
 static StaticQueue_t thermalMgrQueueBuffer;
 static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR_QUEUE_ITEM_SIZE];
 
 static void thermalMgr(void *pvParameters);
 
-void initThermalSystemManager(lm75bd_config_t *config) {
+void initThermalSystemManager(lm75bd_config_t *config)
+{
   memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
   memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
-  
+
   thermalMgrTaskHandle = xTaskCreateStatic(
-    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
-    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+      thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+      config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
 
   memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
   memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
 
   thermalMgrQueueHandle = xQueueCreateStatic(
-    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
-    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
+      THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+      thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
 }
 
-error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event)
+{
   /* Send an event to the thermal manager queue */
-
+  error_code_t errCode;
+  if (!event)
+  {
+    return ERR_CODE_INVALID_ARG;
+  }
+  if (!thermalMgrQueueHandle)
+  {
+    return ERR_CODE_QUEUE_HANDLE_NULL;
+  }
+  if (uxQueueSpacesAvailable(thermalMgrQueueHandle) == 0)
+  {
+    return ERR_CODE_QUEUE_FULL;
+  }
+  LOG_IF_ERROR_CODE(xQueueSend(thermalMgrQueueHandle, event, (TickType_t)0));
   return ERR_CODE_SUCCESS;
 }
 
-void osHandlerLM75BD(void) {
+void osHandlerLM75BD(void)
+{ // interrupt handler
   /* Implement this function */
+  thermal_mgr_event_t event = {.type = THERMAL_MGR_EVENT_OVER_TEMP};
+  thermalMgrSendEvent(&event);
 }
 
-static void thermalMgr(void *pvParameters) {
+static void thermalMgr(void *pvParameters)
+{
   /* Implement this task */
-  while (1) {
-    
+  thermal_mgr_event_t buffer;
+  float temp;
+  error_code_t errCode;
+  while (1)
+  {
+    if (xQueueReceive(thermalMgrQueueHandle, &buffer, (TickType_t)0) == pdFALSE)
+    {
+      LOG_ERROR(ERR_CODE_INVALID_QUEUE_MSG);
```

Comment:
This shouldn't log anything as most of the time thr queue will be empty and it will log errors